analyzer/tests/functional/ctu/test_ctu.py

```diff
@@ -103,6 +104,31 @@ def teardown_method(self, method):
         shutil.rmtree(self.report_dir, ignore_errors=True)
         os.chdir(self.__old_pwd)
 
+    @skipUnlessCTUCapable
+    def test_ctu_non_existing_dir(self):
+        """
+        If the compilation database contains a build action with non-existing
+        directory then analysis shouldn't fail in the pre-analysis phase of
+        CTU.
+        """
+        self.teardown_method(self.test_ctu_non_existing_dir)
+        self.setUpWith('test_files_c', 'buildlog.json', 'reports')
```

Comment:
Not exactly sure why you need these two. Isn't the default setup_method sufficient?

Reply:
`teardown_method()` removes the previous report directory created by some other test function. I need to remove this otherwise `setUpWith()` fails to recreate this directory.

Reply:
That doesn't really answer what I was curious about -- I understand you need `teardown_method()` before manually invoking `setUpWith()`, but isn't the regular `setup_method` that runs automatically good enough? If not, why? Could you put the explanation in the code?

Reply:
Now I understand your comment. I removed these two lines, thank you!